cirq/ops/linear_combinations.py

```diff
@@ -111,3 +111,127 @@ def _pauli_expansion_(self) -> value.LinearDict[str]:
         for gate, coefficient in self.items():
             result += protocols.pauli_expansion(gate) * coefficient
         return result
+
+
+class LinearCombinationOfGateOperations(
```

Comment:
Why is this specific to `GateOperation` instead of `Operation` more generally?

Reply:
Due to fairly limited functionality of bare `Operation`, linear combinations of bare `Operations` wouldn't have much interesting to offer beyond what is already in `LinearDict`, but supporting them would require adding runtime type switches for the interesting things that we can provide for `GateOperations`.

Whenever a linear combination of `Operation`s is needed, one should just use `LinearDict[Operation]`.

(That said, it's very easy to generalize this class to support `Operation` if/when we discover uses for this. I decided to start more concrete and defer generalization.)

Reply:
I don't agree with this.

We have too much code that is specific to GateOperation. There have been a few recent bugs due to this fact.

I don't see any functionality in the class, as it is implemented, that is specific to GateOperation. In particular, you never access the `gate` property!

Reply:
Good point. I wrongly assumed other operation types wouldn't have much to offer in terms of unitaries or Pauli expansions, but in fact they do.

Done.